test/test.js

```diff
@@ -1473,31 +1486,102 @@ test( "don't revalidate the field when pressing special characters", function()
 test( "#1508: Validation fails to trigger when next field is already filled out", function() {
 
 	// The next field is already filled out.
-	$( "#lastname" ).val( "something" );
+	$( "#box2" ).val( "something" );
+
+	var event   = $.Event( "keyup", { keyCode: 9 } ),
+		element = $( "#box1" ),
+		nextE   = $( "#box2" ),
+		validator = $( "#testForm23" ).validate( {
+			rules: {
+				box1: {
+					required: true
+				},
+				box2: {
+					required: true
+				}
+			}
+		} ),
+		check = function( value ) {
+
+			// Fill the first element
+			element.val( "something" );
+
+			// Tab to the next field
+			element.blur();
+			nextE.trigger( event );
+			nextE.focus();
+
+			// Tab back to element
+			nextE.blur();
+			element.trigger( event );
+			element.focus();
 
-	var event     = $.Event( "keyup", { keyCode: 9 } ),
-		element   = $( "#firstname" ),
-		validator = $( "#testForm1" ).validate();
+			// Change the content
+			element.val( value );
+			element.trigger( "keyup" );
+		},
+		eq = function( expected, msg ) {
+			equal( validator.numberOfInvalids(), expected, "There is only one invalid element." );
+			equal( validator.invalidElements()[ 0 ], element[ 0 ], msg );
+		};
+
+	check( "" );
+	eq( 1, "The box1 element should be invalid" );
+} );
+
+test( "[Remote rule] #1508: Validation fails to trigger when next field is already filled out", function( assert ) {
+	assert.expect( 2 );
+
+	// The next field is already filled out.
+	$( "#val3" ).val( "something" );
+
+	var event   = $.Event( "keyup", { keyCode: 9 } ),
+		element = $( "#val2" ),
+		nextE   = $( "#val3" ),
+		done = assert.async(),
+		validator = $( "#testForm24" ).validate( {
+			rules: {
+				val2: {
+					remote: {
+						url: "issue1508.php"
+					}
+				},
+				val3: {
+					required: true
+				}
+			}
+		} ),
+		check = function( value ) {
+
+			// Fill the first element
+			element.val( "something" );
+
+			// Tab to the next field
+			element.blur();
+			nextE.trigger( event );
+			nextE.focus();
 
-	// Fill the first element
-	element.val( "abc" );
+			// Make sure all events will be called before the bellow code
+			setTimeout( function() {
```

Comment:
`setTimeout( ... )` is required here in order to make sure that all events (the first remote call and the rendering) are called before the second call of the remote method.